src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -131,8 +156,59 @@ public Uri GetProxy(Uri uri)
         {
             if (_proxyHelper.ManualSettingsOnly)
             {
+                if ( _bypassLocal)
+                {
+                    IPAddress address = null;
+
+                    if (uri.IsLoopback)
```

Comment:
Does the IsLoopback check force any new allocations under the covers?  IIRC, there's a lot of lazy initialization that happens inside of Uri.

Reply:
It does not seems to. It is bit at Flags and that is set when CreateHostString() is used. Even if this would be first time, it would need to be created later. From that prospective I added as optimization to possibly avoid processing addresses bellow.